test/distances.jl

```diff
@@ -130,8 +123,31 @@ using StringDistances, Unicode, Test
 		@test ismissing(evaluate(Overlap(1), "", missing))
 	end
 
-
-
+	@testset "Differential testing of String, QGramDict, and QGramSortedArray" begin
+		for D in [QGram, Cosine, Jaccard, SorensenDice, Overlap]
+			for _ in 1:100
+				qlen = rand(2:9)
+				dist = D(qlen)
+
+				# 2 random strings with some overlap
+				str1 = randstring(rand(5:10000))
+				ci1 = rand(2:div(length(str1), 2))
+				ci2 = rand((ci1+1):(length(str1)-1))
+				str2 = randstring(ci1-1) * str1[ci1:ci2] * randstring(length(str1)-ci2)
+
+				# QGramDict gets same result as for standard string
+				qd1 = QGramDict(str1, qlen)
+				qd2 = QGramDict(str2, qlen)
+				expected = evaluate(dist, str1, str2)
+				@test expected == evaluate(dist, qd1, qd2)
+
+				# QGramSortedArray gets same result as for standard string
+				qc1 = QGramSortedArray(str1, qlen)
+				qc2 = QGramSortedArray(str2, qlen)
+				@test expected == evaluate(dist, qc1, qc2)
+			end
+		end
```

Comment:
Could you add tests for Unicode characters, as well as arbitrary iterators (i.e. vectors or `graphemes`)?

Reply:
I think so but please see my question on this below...